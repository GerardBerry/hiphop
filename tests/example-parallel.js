var reactive = require("../reactive-kernel.js");
var rjs = require("../xml-compiler.js");
var inspector = require("../inspector.js");

var sigJ = new reactive.Signal("J", false);

var machine = <rjs.reactivemachine name="parallel">
  <rjs.outputsignal ref=${sigJ}/>
  <rjs.localsignal signal_name="I">
    <rjs.parallel>
      <rjs.emit signal_name="I"/>
      <rjs.sequence>
	<rjs.await signal_name="I"/>
	<rjs.emit signal_name="J"/>
      </rjs.sequence>
    </rjs.parallel>
  </rjs.localsignal>
</rjs.reactivemachine>;

exports.prg = machine;
