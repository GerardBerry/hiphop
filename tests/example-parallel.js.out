parallel> ;
--- Output:
parallel> ;
--- Output:
parallel> ;
--- Output:
parallel> ;
--- Output:
